spec/support/shared/shared_block_examples.rb

```diff
@@ -39,3 +39,31 @@
     end
   end
 end
+
+shared_examples 'a fieldset that expects arbitrary blocks of HTML' do
+  context 'when no block is supplied' do
+    subject { builder.send(*args) }
+
+    specify { expect { subject }.to raise_error(LocalJumpError, /no block given/) }
+  end
+
+  context 'when block is supplied' do
+    subject { builder.send(*args, &example_block) }
+
+    let(:example_block) { proc { '<b>some content</b>' } }
+
+    specify { expect { subject }.not_to raise_error }
+    specify { expect { |b| builder.send(*args, &b) }.to yield_control.once }
+
+    context 'with render' do
+      before { allow(helper).to receive(:render) }
+
+      let(:example_block) { proc { helper.render(html: '<b>rendered</b>') } }
+
+      specify do
+        subject
+        expect(helper).to have_received(:render).once
+      end
```

Comment:
I think reordering the setup makes this a tiny bit easier to follow

```suggestion
      let(:example_block) { proc { helper.render(html: '<b>rendered</b>') } }
      
      before { allow(helper).to receive(:render) }
      before { subject }

      specify { expect(helper).to have_received(:render).once }
```

Reply:
Do you prefer to keep the befores separate?

or 
```
before do
  allow(helper).to receive(:render)
  subject
end

Reply:
have applied as separate befores for now as above